daemon/algod/api/server/router.go

```diff
@@ -60,79 +60,75 @@
+	"github.com/labstack/echo/v4"
 
 	"github.com/algorand/go-algorand/daemon/algod/api/server/common"
 	"github.com/algorand/go-algorand/daemon/algod/api/server/lib"
 	"github.com/algorand/go-algorand/daemon/algod/api/server/lib/middlewares"
 	"github.com/algorand/go-algorand/daemon/algod/api/server/v1/routes"
+	"github.com/algorand/go-algorand/daemon/algod/api/server/v2"
+	"github.com/algorand/go-algorand/daemon/algod/api/server/v2/generated"
 	"github.com/algorand/go-algorand/logging"
 	"github.com/algorand/go-algorand/node"
 )
 
 const (
-	apiV1Tag              = "v1"
-	debugRouteName        = "debug"
-	pprofEndpointPrefix   = "/debug/pprof/"
-	urlAuthEndpointPrefix = "/urlAuth/{apiToken:[0-9a-f]+}"
+	apiV1Tag = "v1"
 )
 
-// wrapCtx is used to pass common context to each request without using any
-// global variables.
-func wrapCtx(ctx lib.ReqContext, handler func(lib.ReqContext, http.ResponseWriter, *http.Request)) func(http.ResponseWriter, *http.Request) {
-	return func(w http.ResponseWriter, r *http.Request) {
-		handler(ctx, w, r)
+// wrapCtx passes a common context to each request without a global variable.
+func wrapCtx(ctx lib.ReqContext, handler func(lib.ReqContext, echo.Context)) echo.HandlerFunc {
+	return func(context echo.Context) error {
+		handler(ctx, context)
+		return nil
 	}
 }
 
 // registerHandler registers a set of Routes to [router]. if [prefix] is not empty, it
 // registers the routes to a new sub-router [prefix]
-func registerHandlers(router *mux.Router, prefix string, routes lib.Routes, ctx lib.ReqContext) {
-	if prefix != "" {
-		router = router.PathPrefix(fmt.Sprintf("/%s", prefix)).Subrouter()
-	}
-
+func registerHandlers(router *echo.Echo, prefix string, routes lib.Routes, ctx lib.ReqContext) {
 	for _, route := range routes {
-		r := router.NewRoute()
-		if route.Path != "" {
-			r = r.Path(route.Path)
-		}
-		r = r.Name(route.Name)
-		r = r.Methods(route.Method)
-		r.HandlerFunc(wrapCtx(ctx, route.HandlerFunc))
+		r := router.Add(route.Method, prefix+route.Path, wrapCtx(ctx, route.HandlerFunc))
+		r.Name = route.Name
 	}
 }
 
-// NewRouter builds and returns a new router from routes
-func NewRouter(logger logging.Logger, node *node.AlgorandFullNode, shutdown <-chan struct{}, apiToken string) *mux.Router {
-	router := mux.NewRouter().StrictSlash(true)
+// ConfigureRouter builds and returns a new router from routes
+func ConfigureRouter(logger logging.Logger, node *node.AlgorandFullNode, shutdown <-chan struct{}, apiToken string, e *echo.Echo) {
+	e.Use(echo.WrapMiddleware(middlewares.Logger(logger)))
+	// TODO: New auth middleware - https://github.com/algorand/go-algorand/issues/863
+	e.Use(middlewares.Auth(logger, apiToken))
+	e.Use(echo.WrapMiddleware(middlewares.CORS))
 
-	// Middleware
-	router.Use(middlewares.Logger(logger))
-	router.Use(middlewares.Auth(logger, apiToken))
-	router.Use(middlewares.CORS)
+	// Note: Echo has builtin middleware for logging / CORS that we should investigate:
```

Comment:
it's ok to use any of them, let's just stick to one. 
I'd woudn't change just for the sake of consistency, but don't spend more than 4 minutes on that.